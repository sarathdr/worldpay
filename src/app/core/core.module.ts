import { APP_INITIALIZER, NgModule } from '@angular/core';
import { CommonModule } from '@angular/common';
import { ConfigService } from './services/config.service';
import { PaymentService } from './services/payment.service';
import { CartService } from './services/cart.service';
import { HttpClientModule } from '@angular/common/http';
import { FormsModule } from '@angular/forms';


export const fetchConfig = (appConfig: ConfigService) => {
  return () => appConfig.getPaymentUrls();
};

@NgModule({
  imports: [
    CommonModule,
    FormsModule
  ],
  providers: [
    ConfigService,
    PaymentService,
    HttpClientModule,
    CartService,
    {
      provide: APP_INITIALIZER,
      useFactory: fetchConfig,
      deps: [ConfigService],
      multi: true
    }
  ]
})
export class CoreModule { }
