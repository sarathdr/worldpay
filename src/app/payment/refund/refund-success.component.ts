import { Component, EventEmitter, Output } from '@angular/core';

@Component({
  selector: 'payment-refund-success',
  template: `
    <div class="content">
      <h1>Your payment has been refunded</h1>
      <p>Go to checkout to start again</p>
      <div class="row">
        <button mat-raised-button color="primary" type="button" routerLink="home">Go to home</button>
      </div>
    </div>
  `,
  styles: [
    `
          .content {
              display: flex;
              flex-direction: column;
              justify-content: space-between;
              width: 400px;
              height: 285px;
              margin: 0 auto;
          }

          .row {
              display: flex;
              flex-direction: row;
              justify-content: space-between;
          }
    `
  ]
})
export class RefundSuccessComponent {
  @Output() refundPayment = new EventEmitter<void>();
}
