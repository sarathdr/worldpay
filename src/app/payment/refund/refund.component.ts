import { Component, EventEmitter, Output } from '@angular/core';

@Component({
  selector: 'payment-refund',
  template: `
    <div class="content">
      <h1>Your have successfully made the payment</h1>
      <p>You can still refund the payment</p>
      <div class="row">
        <button mat-raised-button type="button" (click)="refundPayment.emit()">Refund</button>
        <button mat-raised-button color="primary" type="button" routerLink="/home">Go to home</button>
      </div>
    </div>
  `,
  styles: [
    `
          .content {
              display: flex;
              flex-direction: column;
              justify-content: space-between;
              width: 400px;
              height: 285px;
              margin: 0 auto;
          }

          .row {
              display: flex;
              flex-direction: row;
              justify-content: space-between;
          }
    `
  ]
})
export class RefundComponent {
  @Output() refundPayment = new EventEmitter<void>();
}
