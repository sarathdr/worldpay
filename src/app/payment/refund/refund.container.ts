import { Component } from '@angular/core';
import { PaymentService } from '../../core/services/payment.service';
import { Router } from '@angular/router';
import { Subscription } from 'rxjs';
import { take } from 'rxjs/operators';

@Component({
  selector: 'payment-refund-page',
  template: `
    <payment-refund (refundPayment)="refund()" *ngIf="!refundSuccess"></payment-refund>
    <payment-refund-success *ngIf="refundSuccess"></payment-refund-success>
  `
})
export class RefundContainer {
  refundSuccess = false;
  constructor(private paymentService: PaymentService,
              private router: Router) {
  }

  refund() {
    this.paymentService.refundPayment()
      .pipe(take(1))
      .subscribe(() => this.refundSuccess = true);
  }
}
