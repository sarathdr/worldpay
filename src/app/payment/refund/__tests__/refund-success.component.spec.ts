import { TestBed } from '@angular/core/testing';
import { Component } from '@angular/core';
import { MatButtonModule, MatListModule, MatTableModule } from '@angular/material';
import { RouterTestingModule } from '@angular/router/testing';
import { RefundComponent } from '../refund.component';
import { RefundSuccessComponent } from '../refund-success.component';

describe('RefundComponent', () => {

  let fixture;

  beforeEach(() => {
    TestBed.configureTestingModule({
      imports: [
        RouterTestingModule,
        MatListModule,
        MatButtonModule,
        MatTableModule
      ],
      declarations: [
        RefundSuccessComponent,
        TestRefundSuccessComponent
      ],
    });

    fixture = TestBed.createComponent(TestRefundSuccessComponent);
  });

  it('should render the component correctly', () => {
    fixture.detectChanges();
    expect(fixture).toMatchSnapshot();
  });

  @Component({
    selector: 'test-payment-refund-success',
    template: `
      <payment-refund-success></payment-refund-success>
    `
  })
  class TestRefundSuccessComponent {
  }

});
