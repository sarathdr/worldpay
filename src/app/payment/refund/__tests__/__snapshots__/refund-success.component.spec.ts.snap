// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RefundComponent should render the component correctly 1`] = `
<test-payment-refund-success>
  <payment-refund-success>
    <div
      class="content"
    >
      <h1>
        Your payment has been refunded
      </h1>
      <p>
        Go to checkout to start again
      </p>
      <div
        class="row"
      >
        <button
          class="mat-raised-button mat-button-base mat-primary"
          color="primary"
          mat-raised-button=""
          ng-reflect-color="primary"
          ng-reflect-router-link="home"
          routerlink="home"
          tabindex="0"
          type="button"
        >
          <span
            class="mat-button-wrapper"
          >
            Go to home
          </span>
          <div
            class="mat-button-ripple mat-ripple"
            matripple=""
            ng-reflect-centered="false"
            ng-reflect-disabled="false"
            ng-reflect-trigger="[object HTMLButtonElement]"
          />
          <div
            class="mat-button-focus-overlay"
          />
        </button>
      </div>
    </div>
  </payment-refund-success>
</test-payment-refund-success>
`;
